.github/workflows/tests.yml

```diff
@@ -293,6 +293,53 @@ jobs:
           name: qchem-coverage
           path: ./coverage.xml
 
+  data-tests:
+    runs-on: ubuntu-latest

```

Comment:
@obliviateandsurrender I don't really have a strong opinion here, but I'm just curious -- is there a reason against simply having existing tests above install `zstd`, rather than having a separate `data-tests`?

Reply:
Hi Josh! I am not sure I completely understand the alternative you mention. I added separate `data-tests` to make the dependencies (`dill` and `zstd`) independent from the rest of the test suite while also making the tests optional based on their presence for the users. (just like we do for openfermion)

Reply:
(the data tests should be limited to specific python version/s to prevent running them with py3.11)

Reply:
@josh146 is there a pythonic way for `tests/data/test_dataset.py` to install `dill` and `zstd`?